test/functional/blocknotify.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test blocknotify.
+
+Verify that a bitcoind node calls the given command for each block
+"""
+
+import os
+import tempfile
+import time
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class BlockNotifyTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+
+        # get a temporary file path to store the notified blocks
+        fd, path = tempfile.mkstemp(text = True)
```

Comment:
Please use `self.options.tmpdir`, so that files are contained in a single tree for each test.

Also nit: Please no whitespace for named args. (`text=True`)